cmd/influxd/config.go

```diff
@@ -192,8 +192,8 @@ func (c *Config) DataAddrUDP() string {
 }
 
 // DataURL returns the URL required to contact the data server.
-func (c *Config) DataURL() *url.URL {
-	return &url.URL{
+func (c *Config) DataURL() url.URL {
```

Comment:
Using non-pointer URLs is much, much simpler in typical use cases where you want to get a base URL and then add a path. If you do pointers then you need to make a copy so you don't alter the original. With non-pointers it'll do the copy for you and then you don't have to worry about that.